README.md

```diff
@@ -1,2 +1,47 @@
-# rusty-merkle-tree
-Merkle Tree implementation in Rust
+# Merkle Tree
+
+A simple implementation of a Merkle Tree in Rust. This library allows you to create a Merkle Tree, generate proofs for elements, verify elements against the Merkle Tree, and dynamically add new elements.
+
+## Installation
+
+To use this library in your project, add the following line to your `Cargo.toml` file under `[dependencies]`:
+
+```toml
+merkle_tree = { git = "https://github.com/juanimedone/merkle-tree.git", branch = "main" }
+```
+
+## Usage
+
+Here's a basic example of how to use the features of the Merkle Tree library:
+
+```rust
```

Comment:
These are great, but IMO it could be better if the features where described separately. Maybe as bullet points or subheaders, each one with its code snippet. This would allow to quickly see the features and copy the relevant code snippet.

You could also consider having an `examples` directory: more about this [here](http://xion.io/post/code/rust-examples.html).

Reply:
I take your advice, I'll break that section into separate features and I'll add an examples directory with multiple files.